docs/src/main/sphinx/sql/select.md

```diff
@@ -379,6 +379,30 @@ Complex grouping operations are often equivalent to a `UNION ALL` of simple
 does not apply, however, when the source of data for the aggregation
 is non-deterministic.
 
+### AUTO
+
+`AUTO` syntax implicitly groups on every column not included in an aggregation 
```

Comment:
This needs better phrasing. Maybe something like:

> When `AUTO` is specified, the Trino engine automatically determines the grouping columns instead of requiring them to be listed explicitly. In this mode, any column in the `SELECT` list that is not part of an aggregate function is implicitly treated as a grouping column.